### ISSUE TITLE
rocksdbjava : ByteWiseComparatorTest fails with "free(): invalid next size (fast)" when jtest

### ISSUE THREAD

**Issue by guillaumepitel** [facebook/rocksdb#4773]
Normally, **make jtest** compiles rocksjni in DEBUG mode, and all tests are OK. I wanted to run the test with non-debug compilation, and **ByteWiseComparatorTest** fails

### Steps to reproduce the behavior

1 - in **java/Makefile** modify the cmdline below the **run_test:** task to start like this : "java -ea $(JAVA_ARGS) ..."  (this is necessary so that test that must fail on assert actually fail) 
2 - run "**make jtest DEBUG_LEVEL=0**"

You should see a random error after **java_vs_cpp_bytewiseComparator** subtask (i.e. in the **java_vs_java_bytewiseComparator** subtask) about memory corruption, the most frequent I've seen being : "**free(): invalid next size (fast)**"

### Investigations

So I've tried to investigate with valgrind (using this cmdline : 
**valgrind --trace-children=yes --smc-check=all --expensive-definedness-checks=yes java -Djava.compiler=NONE ...** ), see attached file for a possible suspect action (probably not the right suspect, since it's related to port::Mutex). 

[BaseComparator_Valgrind_Suspect.txt](https://github.com/facebook/rocksdb/files/2671037/BaseComparator_Valgrind_Suspect.txt)

NOTE : __I think make jtest should not be run in DEBUG mode by defaut__

**Comment by guillaumepitel** [comment]
I've made more tests. All tests involving Comparators fail in a similar way.

**Comment by adamretter** [comment]
@guillaumepitel did you mean to close this?

**Comment by guillaumepitel** [comment]
> 
> 
> @guillaumepitel did you mean to close this?

Absolutely. Although I tested the problem on the master branch after I found it on my branch, I have probably made the same mistake during the make phase, which created a discrepancy between the librocks.so and rocksjni.so with one in DEBUG and the other in RELEASE

**Comment by adamretter** [comment]
@guillaumepitel Ah I see. So the problem was with your build not with Rocks itself.
